src/components/TimePicker.js

```diff
@@ -0,0 +1,420 @@
+import React, {useEffect, useState, useCallback, useRef, useMemo} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import styles from '../styles/styles';
+import themeColors from '../styles/themes/default';
+import BigNumberPad from './BigNumberPad';
+import Button from './Button';
+import AmountTextInput from './AmountTextInput';
+import * as DeviceCapabilities from '../libs/DeviceCapabilities';
+import useLocalize from '../hooks/useLocalize';
+import CONST from '../CONST';
+import DateUtils from '../libs/DateUtils';
+import Text from './Text';
+import useKeyboardShortcut from '../hooks/useKeyboardShortcut';
+import FormHelpMessage from './FormHelpMessage';
+import refPropTypes from './refPropTypes';
+
+const propTypes = {
+    /** Refs forwarded to the TextInputWithCurrencySymbol */
+    forwardedRef: refPropTypes,
+
+    /** Default value for the inputs */
+    value: PropTypes.string,
+
+    /** Form Error description */
+    errorText: PropTypes.string,
+
+    /** Callback to call when the input changes */
+    onInputChange: PropTypes.func,
+};
+
+const defaultProps = {
+    forwardedRef: null,
+    errorText: '',
+    onInputChange: () => {},
+    value: '',
+};
+
+const AMOUNT_VIEW_ID = 'amountView';
+const NUM_PAD_CONTAINER_VIEW_ID = 'numPadContainerView';
+const NUM_PAD_VIEW_ID = 'numPadView';
+
+function formatHour(hourText) {
+    // // If the integer value of hour is greater than 12, return the second digit with a leading 0
```

Comment:
Some comment tidy-up needed in this function :wink:

Reply:
Thanks 😸